runner/src/server/plugins/engine/views/components/clientsidefileuploadfield.html

```diff
@@ -0,0 +1,390 @@
+  {% endif %}
+{% endmacro %}
+
+
+{% macro ClientSideFileUploadField(component) %}
+
+<head>
+    <script src="/assets/dropzone-min.js"></script>
+    <script type="text/javascript"></script>
+</head>
+
+<style>
+    .dz-template.govuk-table__row {
+        display: table-row;
+    }
+
+    .dz-template.govuk-table__row > .govuk-table__header,
+    .dz-template.govuk-table__row > .govuk-table__cell {
+        display: table-cell;
+        white-space: nowrap;
+    }
+
+    .dz-table__cell.govuk-table__cell:first-child {
+        max-width: 25ch;
+        overflow-wrap: break-word;
+        white-space: normal;
+    }
+
+    .dz-table__cell.govuk-table__cell:not(:first-child) {
+        flex-grow: 1;
+    }
+
+    .dz-progress-bar {
+        display: flex;
+        align-items: center;
+        height: 10px;
+        margin-top: 5px;
+        background-color: #B1B4B6;
+    }
+
+    .dz-progress-bar__value {
+        display: block;
+        height: 100%;
+        background-color: #003078;
+        transition: width 0.2s ease-out;
+        width: 0;
+    }
+
+    .dz-remove {
+        margin-bottom: 0;
+    }
+</style>
+
+<noscript>
+    <style>
+        .requires-js {
+            display: none;
+        }
+    </style>
+</noscript>
+
+{% set is_single_file_upload = component.model.dropzoneConfig.maxFiles === 1 %}
+
+<div id="{{component.model.id}}" class="govuk-form-group {% if component.model.errorMessage %}govuk-form-group--error{% endif %}">
+    {{ govukLabel(component.model.label) }}
+
+    <p class="govuk-error-message">
+      <span class="{% if not component.model.errorMessage %}govuk-visually-hidden{% endif %}">
+          {{ component.model.errorMessage.text | safe }}
+      </span>
+    </p>
+
+    {% if component.model.showNoScriptWarning %}
+    <noscript>
+        <div class="govuk-warning-text" style="margin-bottom: 0;">
+            <span class="govuk-warning-text__icon" aria-hidden="true">!</span>
+            <strong class="govuk-warning-text__text">
+                You need to have JavaScript enabled to see your selected files
+            </strong>
+            <br/>
+            <details class="govuk-details govuk-!-margin-bottom-0" data-module="govuk-details"><summary class="govuk-details__summary">
+                <span class="govuk-details__summary-text">
+                    Why is JavaScript required?
+                </span>
+            </summary><div class="govuk-details__text">
+                <p>
+                    Without JavaScript, you will be able to select files to upload, but you won't be able to see them on this page.
+                </p>
+                <p>
+                    Your files will still be uploaded when you hit "Save and continue". If you revisit this page, you will be able to see your previously uploaded files.
+                </p>
+            </div>
+            </details>
+        </div>
+    </noscript>
+    {% endif %}
```

Comment:
This is wrapped in a condition as I didn't confirm this with design - the JavaScript fallback had to make some compromises on the design as some of the functionality just couldn't be achieved without JavaScript, this is an optional warning to make the user aware of that, will need to confirm with design though, which is why it's off by default.